users/dierkes/preprocessing/wav2vec.py

```diff
@@ -60,8 +79,155 @@ def run(self):
+                torch_nn = wrapped_import("torch.nn.modules")
+                fairseq = wrapped_import("fairseq")
+                wav2vec_models = wrapped_import("fairseq.models.wav2vec")
+                # the following imports are necessary to make sure that these parts are registered also for the wrapped runs
+                audio_pretraining = wrapped_import("fairseq.tasks.audio_pretraining")
+                adam = wrapped_import("fairseq.optim.adam")
+                polynomial_decay = wrapped_import(
+                    "fairseq.optim.lr_scheduler.polynomial_decay_schedule"
+                )
+                wav2vec_criterion = wrapped_import(
+                    "fairseq.criterions.wav2vec_criterion"
+                )
+
+            # Initialize PyTorch example
+            model, cfg, task = fairseq.checkpoint_utils.load_model_ensemble_and_task(
+                [self.wav2vec_config.get_path()]
+            )
+            w2v2_model = model[0]
+            print(
+                "WARNING: Simplifications: feature_grad_mult=1.0, encoder layer dropout=0.0"
+            )
+            w2v2_model.feature_grad_mult = 1.0
+            w2v2_model.encoder.layerdrop = 0.0
+            print(
+                "WARNING: remove quantizer, this involves using non-optimized parameters"
+            )
+            w2v2_model.quantizer = None
+            w2v2_model.project_q.in_features = w2v2_model.embed
+            w2v2_model.project_q._parameters["weight"] = torch.from_numpy(
+                np.random.randn(256, 512).astype("float32")
+            )
+
+            print("WARNING: simplify model by setting mask=False")
+            return w2v2_model(inputs, mask=False, features_only=True)["x"]
+
+        print("Load audio")
+        print("  ", self.audio_input.get_path())
+        audio, fr = soundfile.read(self.audio_input.get_path())
+        audio = np.reshape(audio, [4, len(audio) // 4]).astype(
+            "float32"
+        )  # dim (B, T) -> (4, *)
+        audio_pt = torch.from_numpy(audio)
+        print(f"  input shape: {audio.shape}")
+
+        import pytorch_to_returnn.log
+
+        pytorch_to_returnn.log.Verbosity = 6
+        from pytorch_to_returnn.converter import verify_torch_and_convert_to_returnn
+
+        converter = verify_torch_and_convert_to_returnn(
+            model_func,
+            inputs=audio,
+            inputs_data_kwargs={
+                "shape": (None,),
+                "batch_dim_axis": 0,
+                "time_dim_axis": 1,
+                "feature_dim_axis": None,
+            },
+            returnn_dummy_input_shape=audio.shape,
+            export_tf_checkpoint_save_path=self.returnn_model_out.get_path(),
+            validate_allclose_kwargs={"atol": self.validate_atol}
+            if self.validate_atol
+            else None,
+        )
+        with open(
+            self.returnn_model_out.get_path() + ".network.dict", "wt", encoding="utf-8"
```

Comment:
```suggestion
            self.out_returnn_config.get_path(), "wt", encoding="utf-8"
```
Or similar, but directly using the output path for the config, see comment above.